src/main/scala/mesosphere/mesos/ResourceMatcher.scala

```diff
@@ -51,20 +87,28 @@ object ResourceMatcher {
       badConstraints.isEmpty
     }
 
-    def portsOpt: Option[Seq[RangesResource]] = new PortsMatcher(app, offer, acceptedResourceRoles).portRanges match {
-      case None =>
-        log.warn("App ports are not available in the offer.")
-        None
+    def portsOpt: Option[Seq[RangesResource]] = new PortsMatcher(app, offer, acceptedResourceRoles).portRanges
+
+    val basicResourceMatches = Map(
+      "cpu" -> cpuMatchOpt,
+      "disk" -> diskMatchOpt,
+      "mem" -> memMatchOpt
+    )
 
-      case x @ Some(portRanges) =>
-        log.debug("Met all constraints.")
-        x
+    if (!basicResourceMatches.values.forAll(_.map(_.matches).getOrElse(false))) {
```

Comment:
This line doesn't compile anymore.

The previous (correct) version was:

`if (basicResourceMatches.values.exists(_.exists(!_.matches))) {`